Source/Scene/Cesium3DTileset.js

```diff
@@ -0,0 +1,1821 @@
+
+        /**
+         * This property is for debugging only; it is not optimized for production use.
+         * <p>
+         * When true, draws labels to indicate the geometric error of each tile.
+         * </p>
+         *
+         * @type {Boolean}
+         * @default false
+         */
+        this.debugShowGeometricError = defaultValue(options.debugShowGeometricError, false);
+
+        /**
+         * This property is for debugging only; it is not optimized for production use.
+         * <p>
+         * When true, draws labels to indicate the number of commands, points, triangles and features of each tile.
+         * </p>
+         *
+         * @type {Boolean}
+         * @default false
+         */
+        this.debugShowRenderingStatistics = defaultValue(options.debugShowRenderingStatistics, false);
+
+        /**
+         * This property is for debugging only; it is not optimized for production use.
+         * <p>
+         * When true, draws labels to indicate the geometry and texture memory usage of each tile.
+         * </p>
+         *
+         * @type {Boolean}
+         * @default false
+         */
+        this.debugShowMemoryUsage = defaultValue(options.debugShowMemoryUsage, false);
+
+        var that = this;
+
+        // We don't know the distance of the tileset until tileset.json is loaded, so use the default distance for now
+        RequestScheduler.request(tilesetUrl, loadJson, undefined, RequestType.TILES3D).then(function(tilesetJson) {
+            if (that.isDestroyed()) {
+                return when.reject('tileset is destroyed');
+            }
+            that._root = that.loadTileset(tilesetUrl, tilesetJson);
+
+            var gltfUpAxis = defined(tilesetJson.asset.gltfUpAxis) ? Axis.fromName(tilesetJson.asset.gltfUpAxis) : Axis.Y;
+            that._asset = tilesetJson.asset;
+            that._properties = tilesetJson.properties;
+            that._geometricError = tilesetJson.geometricError;
+            that._gltfUpAxis = gltfUpAxis;
+            that._readyPromise.resolve(that);
+        }).otherwise(function(error) {
+            that._readyPromise.reject(error);
+        });
+    }
+
+    defineProperties(Cesium3DTileset.prototype, {
+        /**
+         * Gets the tileset's asset object property, which contains metadata about the tileset.
+         * <p>
+         * See the {@link https://github.com/AnalyticalGraphicsInc/3d-tiles/blob/master/schema/asset.schema.json|asset schema}
+         * in the 3D Tiles spec for the full set of properties.
+         * </p>
+         *
+         * @memberof Cesium3DTileset.prototype
+         *
+         * @type {Object}
+         * @readonly
+         *
+         * @exception {DeveloperError} The tileset is not loaded.  Use Cesium3DTileset.readyPromise or wait for Cesium3DTileset.ready to be true.
+         *
+         * @example
+         * console.log('3D Tiles version: ' + tileset.asset.version);
```

Comment:
Sorry, getting crabby after such a long PR.  What I meant to say is these types of examples aren't very beneficial because they are too simplistic.  Better to just leave them out.